### DISCUSSION TITLE
Where can I learn JavaScript in order to contribute?

### DISCUSSION THREAD

**Discussion by just-a-hriday** [ScratchAddons/ScratchAddons#2869] [Help] [answered]
With the number of ideas I have, I think I should stop creating issues and start creating pull requests and doing the changes myself. 

But the thing is, I don't know where to learn JavaScript. 
HTML and CSS I can learn - but I don't have the slightest idea of how to learn JavaScript.

**Comment by TheChorister** [comment]
I used [w3schools](https://w3schools.com) for learning web development. ([go here](https://w3schools.com/js/default.asp) to learn JavaScript specifically)

**Reply by just-a-hriday** [reply]
Thanks! I'll try that out :wink:

**Comment by mxmou** [comment, marked as answer]
[MDN](https://developer.mozilla.org/en-US/docs/Learn/JavaScript) is probably the best place to learn any web technology. It has much less missing or outdated information than W3Schools.

**Reply by just-a-hriday** [reply]
Thanks! I'll try that out 😉

**Comment by cobaltt7** [comment]
I like the Web Dev Simplified YouTube channel. He has a lot of videos, and also some paid courses.
________________________________

**Reply by WorldLanguages** [reply]
[Fireship](https://www.youtube.com/channel/UCsBjURrPoezykLs9EqgamOA) also has some cool videos

**Reply by TheColaber** [reply]
Yeah, I personally use/find channels like these, when I want to do something specific, like learn svg paths. You first need to learn the basics like data types and what the document is, and then try simple task, like querying an element and replacing it's text. That will help you get started.

**Comment by daniel4-scratch** [comment]
I would book for some tutoring. It is really helpful with a person right there to ask questions and get a response in 1 second (instead of waiting 1 day for a response). And I easily learn the basics straight away. I use some tutoring for python and it really worked and I was in a class. So I could ask people around me too.

But I self-taught myself javascript, html and css. And w3schools is really awesome coz it taught me a lot of stuff.

If you are really stuck ask people on stack overflow.
